jvm/src/common/Language/Java/Unsafe.hs

```diff
@@ -558,7 +558,7 @@ reifyMVector mk finalize jobj0 = do
     n <- getArrayLength jobj
     ptr <- mk jobj
     fptr <- newForeignPtr ptr $ finalize jobj ptr
-                                  `finally` deleteGlobalRefNonFinalized jobj
+                                  `finally` runInAttachedThread (deleteGlobalRefNonFinalized jobj)
```

Comment:
This is well spotted. Note though, that `finalize jobj ptr` might do jni calls, so it needs to be wrapped as well.
```
newForeignPtr ptr $ runInAttachedThread $
  finalize jobj ptr `finally` deleteGlobalRefNonFinalized jobj
```

Also, this finalizer is only used for vectors. The finalizer of global refs in Foreign/JNI/Unsafe.hs hasn't been updated.